File: sunnad-ios/sunnad-ios/Domain/Repositories/QuotesRepository.swift
import Foundation

protocol QuotesRepository: Sendable {
    func fetchQuoteOfDay(locale: String, on day: Date, calendar: Calendar, timeZone: TimeZone) async throws -> Quote?
    func fetchSavedQuotes() async throws -> [SavedQuote]
    func saveQuote(_ quote: Quote, savedAt: Date) async throws
    func deleteAllSavedQuotes() async throws
}

File: sunnad-android/app/src/main/java/com/arystan/almasuly/sunnadandroid/domain/repository/QuotesRepository.kt
package com.arystan.almasuly.sunnadandroid.domain.repository

import com.arystan.almasuly.sunnadandroid.core.model.Quote
import com.arystan.almasuly.sunnadandroid.core.model.SavedQuote
import java.time.LocalDate
import java.util.UUID

interface QuotesRepository {
    suspend fun fetchQuoteOfDay(locale: String, day: LocalDate): Quote?
    suspend fun fetchSavedQuotes(): List<SavedQuote>
    suspend fun isQuoteSaved(quoteId: UUID): Boolean
    suspend fun saveQuote(quote: Quote)
    suspend fun deleteAllSavedQuotes()
}
